### DISCUSSION TITLE
vue版本ant-design-vue DatePicker组件不显示

### DISCUSSION THREAD

**Discussion by yangshuwei** [alibaba/formily#1468] [Q&A 2.x] [answered]
{
  type: 'object',
  properties: {
    birthday: {
      type: 'string',
      required: true,
      title: '生日',
      'x-component': 'DatePicker',
      'x-decorator': 'FormItem',
      'x-component-props':{
        format:"YYYY-MM-DD HH:mm",
        placeholder:"['Start Time', 'End Time']"
      }
    },
  },
}

**Comment by janryWang** [comment]
你不觉得这种问题很没营养吗？第一，代码格式有问题，第二，没有复现地址，能不能多考虑下开源维护者的感受

**Comment by MisicDemone** [comment]
提供的信息太少无法复现出该问题，参考：https://codesandbox.io/s/heuristic-kapitsa-omgwv?file=/src/App.vue

**Reply by MisicDemone** [reply]
@yangshuwei 另外建议fork一份这个codesandbox，后续提问能作为复现方式带上。

**Reply by yangshuwei** [reply]
谢谢解答啊，不知道是我版本的问题还是什么原因？组件始终加载不出来 [https://codesandbox.io/s/black-sunset-k0v8e?file=/src/components/HelloWorld.vue](url)

**Reply by yangshuwei** [reply]
vue3.0 ant-design-vue 2.0

**Comment by MisicDemone** [comment, marked as answer]
``` ts
const DatePicker = (props, context) => h(ADatePicker, context.attrs);
```

当前将DatePicker包裹在一个没有children的函数式组件中可以解决这个问题。

产生原因是 [DatePicker的实现](https://github.com/vueComponent/ant-design-vue/blob/next/components/date-picker/createPicker.tsx#L258)中，默认的input组件能被插槽覆盖，而formily-vue始终为组件添加了默认插槽，但是这个插槽可能是空的。后续版本会提供更严谨的插槽支持能从根源解决该问题。
